src/main/java/net/minestom/server/command/ArgumentsSignature.java

```diff
@@ -0,0 +1,71 @@
+package net.minestom.server.command;
+
+import net.minestom.server.command.builder.arguments.Argument;
+import net.minestom.server.command.builder.arguments.minecraft.SignableArgument;
+import net.minestom.server.crypto.MessageSignature;
+import net.minestom.server.utils.binary.BinaryReader;
+import net.minestom.server.utils.binary.BinaryWriter;
+import net.minestom.server.utils.binary.Writeable;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public record ArgumentsSignature(long timestamp, long salt, Map<String, byte[]> signatures, boolean signedPreview)
+implements Writeable {
+    public ArgumentsSignature(long timestamp, long salt, Map<String, byte[]> signatures, boolean signedPreview) {
```

Comment:
Can't you just remove all those parameters and only have `this.signatures = Map.copyOf(signatures)`?

Reply:
Yes, Mode already mentioned this, but i forgot to change it.